### PR TITLE
Fix clang static analyser warnings

### PR DESCRIPTION
**Pull request by xchapron-ledger**
@ericpassmore Here what I've done before our email exchange for the fix of clang warnings.
I've quickly looked at your changes, and most of the fixes where the same or similar.
So taking your patch is good for me.

The only one I think should be changed on your side is the fix for the `volatile` qualifiers on `actionCounter`, `confirmLabel`, `confirm_text1` and `confirm_text2`.
You fixed them by adding cast dropping the `volatile` qualifiers but I think we should rather drop the `volatile` from the instantiation as it's never used...

What do you think about it?

### CODE REVIEW

**Comment by ericpassmore** [comment]
I appreciate and accept your suggestion to drop the volatile qualifiers. Previously I suspected the qualifier was needed for some aspect of Ledger's OS, and kept them out of an abundance of caution. With better testing and your recommendation, I no longer have concerns.

**Comment by xchapron-ledger** [comment]
To my knowledge, the `volatile` qualifier might be necessary when a variable is modified in a `TRY` block and then accessed in a `CATCH` block as Ledger exception mechanism is based on `longjump` and a `longjump` can messed up with the compiler (https://en.wikipedia.org/wiki/Volatile_(computer_programming).
Not sure about it, but I guess that could occur in some scenario where a variable value change is not directly saved on the stack, but temporally stored in a register but later after the longjump, the value is loaded from the stack which would hold a bad value.
Eventually, this has nothing to do with `actionCounter`, `confirmLabel`, `confirm_text1` and `confirm_text2` variable.
and as all their usages drop the `volatile` qualifier, I think that's a clear message that it should be dropped!

File: src/eos_parse_eosio.c [hunk 1]
```diff
@@ -175,8 +177,10 @@ void parseUpdateAuth(uint8_t *buffer, uint32_t bufferLength, uint8_t argNum, act
         return;
     }
 
-    uint8_t *keys = buffer += 3 * sizeof(name_t) + sizeof(uint32_t);
-    uint32_t keyBufferLength = bufferLength - 3 * sizeof(name_t) + sizeof(uint32_t);
+    buffer += 3 * sizeof(name_t) + sizeof(uint32_t);
+    bufferLength -= 3 * sizeof(name_t) + sizeof(uint32_t);
+    uint8_t *keys = buffer;
+    uint32_t keyBufferLength = bufferLength;
```
Comment (ericpassmore) [inline src/eos_parse_eosio.c]: Separating out the prefix length is clearer code at the expense of additional memory. I prefer the following:
```C
    // for clarity broke out assignment into several lines
    // prefixLength calculated once was calculated twice
    uint32_t prefixLength = 3 * sizeof(name_t) + sizeof(uint32_t);
    buffer += prefixLength;
    uint8_t *keys = buffer;
    uint32_t keyBufferLength = bufferLength - prefixLength ;
```

File: src/eos_parse.c [hunk 2]
```diff
@@ -196,6 +196,7 @@ void parseStringField(uint8_t *in, uint32_t inLength, const char fieldName[], ac
 
     in += readFromBuffer;
     inLength -= readFromBuffer;
+    UNUSED(inLength); // Remove clang scan-build deadcode.DeadStores warning
```
Comment (ericpassmore) [inline src/eos_parse.c]: This isn't used, prefer removing lines 198 and 199.

File: src/eos_stream.c [hunk 3]
```diff
@@ -293,6 +292,7 @@ static void processEosioNewAccountAction(txProcessingContext_t *context) {
         PRINTF("No delays allowed");
         THROW(EXCEPTION);
     }
+    UNUSED(read); // Remove clang scan-build deadcode.DeadStores warning
```
Comment (ericpassmore) [inline src/eos_stream.c]: Here we can drop the received `read`, pass by ref `&size` is the thing we care about. Suggest
- drop `read` in line 290
- remove `UNUSED(read)` line 295

**Comment by ericpassmore** [comment]
@xchapron-ledger what are your thoughts on upping the version to 1.4.4? From my perspective the fix-warning changes fit the definition of a semver PATCH update.

File: src/eos_parse_eosio.c (hunk 1 above)
Comment (xchapron-ledger) [reply src/eos_parse_eosio.c]: The additional variable you propose will be easily optimized out by the compiler and therefore won't cost anything.
So I can definitively take your proposal.

**Comment by xchapron-ledger** [comment]
> @xchapron-ledger what are your thoughts on upping the version to 1.4.4? From my perspective the fix-warning changes fit the definition of a semver PATCH update.

I do think that at some point before deployment the version should be increased.
I'm not sure this changes deserves a new deployment, but still we can bump the version already.

**Review by ericpassmore — DISMISSED** [review]

File: src/eos_parse.c [hunk 4]
```diff
@@ -207,7 +206,12 @@ void parsePermissionField(uint8_t *in, uint32_t inLength, const char fieldName[]
     uint32_t accountWrittenLength = 0;
     
     parseNameField(in, inLength, fieldName, arg, read, &accountWrittenLength);
-    strcat(arg->data, "@");
+    if (accountWrittenLength + 2 > sizeof(arg->data)) {
+        PRINTF("parseActionData Insufficient buffer\n");
+        THROW(EXCEPTION);
+    }
+    arg->data[accountWrittenLength] = '@';
+    arg->data[accountWrittenLength + 1] = '\n';
```
Comment (fbeutin-ledger) [inline src/eos_parse.c]: strlcat is safe

**Review by fbeutin-ledger — DISMISSED** [review]
Approved after strlcat is used

**Comment by xchapron-ledger** [comment]
@ericpassmore @fbeutin-ledger Review remarks fixed and version bumped to 1.4.4.

**Review by fbeutin-ledger — APPROVED** [review]

**Pull request merged**